java/ql/src/experimental/Security/CWE/CWE-601/SpringUrlRedirect.qll

```diff
@@ -71,3 +76,66 @@ class SpringUrlRedirectSink extends DataFlow::Node {
     )
   }
 }
+
+/**
+ * A sink for URL Redirection via the `ResponseEntity` class.
+ */
+private class SpringResponseEntityUrlRedirectSink extends SpringUrlRedirectSink {
+  SpringResponseEntityUrlRedirectSink() {
+    // Find `new ResponseEntity<>(httpHeaders, ...)` or
+    // `new ResponseEntity<>(..., httpHeaders, ...)` sinks
+    exists(ClassInstanceExpr cie, Argument argument |
+      cie.getConstructedType() instanceof SpringResponseEntity and
+      argument.getType() instanceof SpringHttpHeaders and
+      (argument = cie.getArgument(0) or argument = cie.getArgument(1)) and
+      this.asExpr() = argument
+    )
+    or
+    // Find `ResponseEntity.status(...).headers(taintHeaders).build()` or
+    // `ResponseEntity.status(...).location(URI.create(taintURL)).build()` sinks
+    exists(MethodAccess ma, ResponseEntityExpr e |
+      ma = e.getAChildExpr() and
```

Comment:
Why are you asking for the `headers` or `location` calls to be a direct child of something of type `SpringResponseEntity`? When is a `headers` or `location` argument not a sink?